subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testng/TestNGMultiVersionTest.groovy

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.testng
+
+import org.gradle.integtests.fixtures.MultiVersionIntegrationSpec
+import org.gradle.testing.fixture.AbstractJUnitMultiVersionIntegrationTest
+
+trait TestNGMultiVersionTest {
+    @Override
+    AbstractJUnitMultiVersionIntegrationTest.BuildScriptConfiguration getBuildScriptConfiguration() {
+        return new TestNGBuildSourceConfiguration()
+    }
+
+    @Override
+    AbstractJUnitMultiVersionIntegrationTest.TestSourceConfiguration getTestSourceConfiguration() {
+        return new TestNGTestSourceConfiguration()
+    }
+
+    static class TestNGBuildSourceConfiguration implements AbstractJUnitMultiVersionIntegrationTest.BuildScriptConfiguration {
+        @Override
+        String getTestFrameworkDependencies(String sourceSet) {
+            return """
+                    testImplementation 'org.testng:testng:${MultiVersionIntegrationSpec.version}'
```

Comment:
Shouldn't this be handling the `sourceSet`?

Reply:
Yep, good catch.  Fixed.